pandas/core/aggregation.py

```diff
@@ -5,14 +5,72 @@
 
 from collections import defaultdict
 from functools import partial
-from typing import Any, DefaultDict, List, Sequence, Tuple
+from typing import Any, DefaultDict, List, Optional, Sequence, Tuple, Union
 
 from pandas.core.dtypes.common import is_dict_like, is_list_like
+from pandas.core.dtypes.generic import ABCDataFrame
 
+from pandas.core.base import SpecificationError
 import pandas.core.common as com
 from pandas.core.indexes.api import Index
 
 
+def reconstruct_func(
+    func: Optional[Union[List[Any], dict]], *args, **kwargs: dict
+) -> Tuple[bool, Any, Optional[List[str]], Optional[List[int]]]:
+    """
+    This is the internal function to reconstruct func given if there is relabeling
+    or not and also normalize the keyword to get new order of columns.
+
+    If relabeling is True, will return relabeling, reconstructed func, column
+    names, and the reconstructed order of columns.
+    If relabeling is False, the columns and order will be None.
+
+    Parameters
+    ----------
+    func: aggregated function
+    **kwargs: dict
```

Comment:
Can you document what these are used for?

Reply:
added a short sentence